packages/orbit-components/src/InputSelect/index.tsx

```diff
@@ -204,27 +205,33 @@ const InputSelect = React.forwardRef<HTMLInputElement, Props>(
         error={isLargeMobile && error}
         label={isLargeMobile && label}
         disabled={disabled}
-        onFocus={handleFocus}
+        onFocus={!readOnly ? handleFocus : undefined}
```

Comment:
Thinking about this, a `readOnly` input is focusable. So we still need to call `handleFocus` here, because there might be some `onFocus` defined. What we need to do is prevent the options to be shown.